mesonbuild/modules/unstable_wayland.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2022 Mark Bolhuis <mark@bolhuis.dev>
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from . import ExtensionModule, ModuleReturnValue
+from .. import mlog
+from ..build import CustomTarget
+from ..interpreter.type_checking import NoneType
+from ..interpreterbase import FeatureNew, typed_pos_args, typed_kwargs, KwargInfo
+from ..mesonlib import File, MesonException
+
+
+class WaylandModule(ExtensionModule):
+
+    @FeatureNew('wayland module', '0.62.0')
+    def __init__(self, interpreter):
+        super().__init__(interpreter)
+
+        self.protocols_dep = None
+        self.pkgdatadir = None
+        self.scanner_bin = None
+
+        self.methods.update({
+            'scan_xml': self.scan_xml,
+            'find_protocol': self.find_protocol,
+        })
+
+    @typed_pos_args('wayland.scan_xml', varargs=(str, File), min_varargs=1)
+    @typed_kwargs(
+        'wayland.scan_xml',
+        KwargInfo('side', str, default='client'),
+        KwargInfo('scope', str, default='private'),
```

Comment:
Can do even better: `KwargInfo('side', str, default='client', validator=in_set_validator({'client', 'server'})),` likewise for scope. With that you can remove the check below.